oci/pull.bzl

```diff
@@ -37,24 +37,164 @@ oci_image(
 """
 
 load("@aspect_bazel_lib//lib:paths.bzl", "BASH_RLOCATION_FUNCTION")
+load("@aspect_bazel_lib//lib:base64.bzl", "base64")
+load("@aspect_bazel_lib//lib:repo_utils.bzl", "repo_utils")
+
+def _strip_host(url):
+    # TODO: a principled way of doing this
+    return url.replace("http://", "").replace("https://", "").replace("/v1/", "")
+
+def _file_exists(rctx, path):
+    result = rctx.execute(["stat", path])
```

Comment:
feels like a portability concern, isn't there a way to detect this in the starlark interpreter without any subprocess?

Reply:
nothing I could find. afaik nothing in the rctx can tell whether a file exists.

Reply:
maybe @gregmagolan knows as he has done a lot of repository rule stuff

Reply:
Calling `rctx.path()` on a non-existent file will just fail. Although you also have to be careful with using `exists` since calling `rctx.path` on a dynamic label (a label that was _not_ passed in via an attribute but rather constructed dynamically) will cause the repository rule to restart so its best to do all of these dynamical label lookups at the beginning of the repository rule before you do any work that takes time or else you'll end up doing that work multiple times to fetch the repository

Reply:
yeah. It would be great to put that helper somewhere in bazel-lib.